src/test/scala/org/apache/spark/sql/pulsar/PulsarSourceTest.scala

```diff
@@ -126,6 +126,71 @@ class PulsarSourceTest extends StreamTest with SharedSparkSession with PulsarTes
       s"AddPulsarData(topics = $topics, data = $data, message = $message)"
   }
 
+  /**
+   * Add data to Pulsar with partition specified
+   *
+   * `topicAction` can be used to run actions for each topic before inserting data.
+   */
+  case class AddPulsarDataWithPartition(
+    topics: Set[String],
```

Comment:
This interface look weird, I guess you want to send to a specific topic partition, why is the parameter a set of topics?

Reply:
Good point, will refactor